### PR TITLE
Anna/rework geometry groups

### PR DESCRIPTION
**Pull request by Verkhovskaya**

### CODE REVIEW

File: src/geometry/TorusSegmentGroup.ts [hunk 1]
```diff
@@ -152,7 +103,7 @@ export default class TorusSegmentGroup extends PlaneGroup {
         computeCircleBoundingBox(
           tubeCenter,
           tubeNormal,
-          this.getTubeRadius(index),
+          this.data.getNumber('radiusB', index),
```
Comment (anderscognite) [inline src/geometry/TorusSegmentGroup.ts]: I think `tubeRadius` is more readable/understandable than just `radiusB` on torus?

File: src/geometry/TorusSegmentGroup.ts [hunk 2]
```diff
@@ -142,7 +93,7 @@ export default class TorusSegmentGroup extends PlaneGroup {
       tubeCenter
         .set(x * radius, y * radius, 0)
         .applyQuaternion(secondRotation)
-        .add(this.getCenter(vector2, index))
+        .add(this.data.getVector3('centerA', vector2, index))
```
Comment (anderscognite) [inline src/geometry/TorusSegmentGroup.ts]: Why do we have `centerA` when there is only one center?

File: src/geometry/TorusSegmentGroup.ts [hunk 3]
```diff
@@ -95,29 +42,33 @@ export default class TorusSegmentGroup extends PlaneGroup {
...
-    this.setRadius(radius, this.count);
-    this.setTubeRadius(tubeRadius, this.count);
-    this.setAngle(angle, this.count);
-    this.setArcAngle(arcAngle, this.count);
-    this.count += 1;
+    this.setNodeId(nodeId, this.data.count);
+    this.setTreeIndex(treeIndex, this.data.count);
+    this.setColor(color, this.data.count);
+    this.data.add({
+      nodeId: nodeId,
```
Comment (anderscognite) [inline src/geometry/TorusSegmentGroup.ts]: When the variable you set as value is named the same as the key, you can simply do
```
this.data.add({
      nodeId,
      ...
});

File: src/geometry/SphericalSegmentGroup.ts [hunk 4]
```diff
@@ -137,14 +75,14 @@ export default class SphericalSegmentGroup extends PrimitiveGroup {
...
       .applyMatrix3(normalMatrix)
       .normalize();
     const scaling = matrix.getMaxScaleOnAxis();
-    const radius = scaling * this.getRadius(index);
-    const height = scaling * this.getHeight(index);
+    const radius = scaling * this.data.getNumber('radiusA', index);
+    const height = scaling * this.data.getNumber('heightA', index);
 
-    sphereCenter.copy(this.getCenter(globalCenter, index)).applyMatrix4(matrix);
+    sphereCenter.copy(this.data.getVector3('centerA', globalCenter, index)).applyMatrix4(matrix);
```
Comment (anderscognite) [inline src/geometry/SphericalSegmentGroup.ts]: We should call it just `center` here too, no?

File: src/geometry/GeometryGroupDataParameters.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+rotationAngle?: any;
+triangleCount?: any;
+vertex1?: any;
+vertex2?: any;
+vertex3?: any;
+vertex4?: any;
+}
+
+export const int64Properties: propertyName[] = ['nodeId'];
+export const int32Properties: propertyName[] =
```
Comment (anderscognite) [inline src/geometry/GeometryGroupDataParameters.ts]: All of these should be float32Properties (not int), same goes for `nodeId` as there is no int64 in js :D

**Review by anderscognite — CHANGES_REQUESTED** [review]
This is SUPER nice! I guess a few changes are required:
1) call them float32properties and not int (most of them are floating point numbers)
2) maybe we shouldn't use `centerA` when there is only one center? The weirdest one is `radiusA` and `radiusB` instead of `radius` and `tubeRadius`. However, comments could solve the latter.
3) We need the attributes (maybe an isAttribute = true/false) per property?

File: src/geometry/GeometryGroupDataParameters.ts (hunk 5 above)
Comment (Verkhovskaya) [reply src/geometry/GeometryGroupDataParameters.ts]: Fixed

File: src/geometry/SphericalSegmentGroup.ts (hunk 4 above)
Comment (Verkhovskaya) [reply src/geometry/SphericalSegmentGroup.ts]: Fixed

File: src/geometry/TorusSegmentGroup.ts (hunk 2 above)
Comment (Verkhovskaya) [reply src/geometry/TorusSegmentGroup.ts]: Fixed

File: src/geometry/TorusSegmentGroup.ts (hunk 1 above)
Comment (Verkhovskaya) [reply src/geometry/TorusSegmentGroup.ts]: Fixed

File: src/geometry/TorusSegmentGroup.ts (hunk 3 above)
Comment (Verkhovskaya) [reply src/geometry/TorusSegmentGroup.ts]: Fixed

**Pull request merged**